test/portable_link_flags_darwin/test.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+set -e
+
+$1
```

Comment:
Could you instead run `otool -L "$1"` and verify (by grep or whatever) that it doesn't contain "libstd" or "libtest"?

I want us to be sure that we're actually statically linking stuff, rather than setting up rpaths or whatever so that in the bazel-bin tree the binary _happens_ to work but if you moved it somewhere else or copied it to another machine it would fail to.

You can use `target_compatible_with` to make sure this test only runs on darwin where `otool` will be present.